packages/p2p/src/pages/advertiser-page/advertiser-page.jsx

```diff
@@ -75,14 +75,30 @@ const AdvertiserPage = () => {
 
     const { data: p2p_advert_info } = useP2PAdvertInfo(counterparty_advert_id);
 
-    const showErrorModal = () => {
+    const showErrorModal = eligibility_status => {
+        let error_message = localize("It's either deleted or no longer active.");
+        let error_modal_title = localize('This ad is unavailable');
+
+        if (eligibility_status?.length > 0) {
+            error_modal_title = '';
+            if (eligibility_status.length === 1) {
+                if (eligibility_status.includes('completion_rate')) {
+                    error_message = localize('Your completion rate is too low for this ad.');
+                } else if (eligibility_status.includes('join_date')) {
+                    error_message = localize("You've not used Deriv P2P long enough for this ad.");
+                }
+            } else {
+                error_message = localize("The advertiser has set conditions for this ad that you don't meet.");
+            }
+        }
```

Comment:
this check is similar to the one in BuySellActionRow component below, could we perhaps extract it out into a utils file and import it as a function, we just need to pass the eligibility status